common/persistence/sql/sqlExecutionManager.go

```diff
@@ -609,26 +623,37 @@ func (m *sqlExecutionManager) updateWorkflowExecutionTx(tx sqldb.Tx, request *p.
 		}
 	} else {
 		executionInfo := request.ExecutionInfo
-		startVersion := common.EmptyVersion
-		lastWriteVersion := common.EmptyVersion
-		if request.ReplicationState != nil {
-			startVersion = request.ReplicationState.StartVersion
-			lastWriteVersion = request.ReplicationState.LastWriteVersion
-		}
-		// this is only to update the current record
-		if err := assertAndUpdateCurrentExecution(tx,
-			shardID,
-			domainID,
-			workflowID,
-			runID,
-			runID,
-			executionInfo.CreateRequestID,
-			executionInfo.State,
-			executionInfo.CloseStatus,
-			startVersion,
-			lastWriteVersion); err != nil {
+		switch executionInfo.State {
+		case p.WorkflowStateZombie:
+			if err := assertNotCurrentExecution(tx, shardID, domainID, workflowID, runID); err != nil {
```

Comment:
Can this NotCurrentExecution check move to executionStore so that we don't need to have them in both cass and sql

Reply:
sql and cassandra side use different ways to check whether the workflow to be updated is the current workflow

sql actually have the current record, in persistenc logic (while doing create / update), while cassandra does not.